sections/appendix/uppaal.tex

```diff
@@ -0,0 +1,141 @@
+\section{Understanding a \uppaal model}\label{app:uppaal}
+
+The goal of this appendix is not to give an extensive guide on how to use \uppaal, but rather to serve as a short introduction on how to read the models described in this report.
+For a more in-depth description of how \uppaal works, we recommend the official tutorial for \uppaal 4.0 by Gerd Behrmann, Alexandre David and Kim G. Larsen \cite{uppaaltutorial}.
+
+\subsection{What is \uppaal?}
+\uppaal is an integrated tool environment for modeling and verification of timed automata.
+It is mainly used for simulating and verifying models of real-time systems \cite{uppaalintro} by means of property checking, where it can automatically generate a diagnostic trace to explain why a given property is, or is not, satisfied by the modeled system.\\
+This property checking is done using a reachability analysis, meaning that it will search through the state space of the system to find a state where a given property is true.
+Using property checking it is then possible to check for properties like whether the system can end up in a deadlock - or in the context of this system, it would be possible to check whether it is possible that one client is still in the lobby while the other clients are receiving in-game information such as positional data.
+
+\subsection{Format of a system in \uppaal}
+A very basic model in \uppaal consists of a set of locations and edges as seen on \autoref{fig:app:uppaal:locations}.
+In this basic model, we have two locations: \uppLoc{A} and \uppLoc{B} and two transitions: From \uppLoc{A} to \uppLoc{B}, and from \uppLoc{B} to \uppLoc{A}.
+The location \uppLoc{A} has an extra ring, which indicates that it is an initial location, meaning that this is where the system will start when initialized.
+
+\begin{figure}[H]
+    \centering
+    \includegraphics[width=0.6\linewidth]{uppaal/applocations.png}
+    \caption{Locations in \uppaal.}
+    \label{fig:app:uppaal:locations}
+\end{figure}
+
+This visual model combined with a set of local declarations related to the model is called a template.
+
+In addition to the local declarations, which are scoped to the current template, it is also possible to have global declarations (First object in the Project folder on \autoref{fig:app:uppaal:templates}), which are variables and functions that are shared across the entire system.
+
+\begin{figure}[H]
+    \centering
+    \includegraphics[width=0.6\linewidth]{uppaal/apptemplates.png}
+    \caption{Content of a \uppaal project.}
+    \label{fig:app:uppaal:templates}
+\end{figure}
+
+When the system is started in the simulator, these \uppTemp{templates} will be instantiated as \uppProc{processes}, which start in their initial locations.
+
+From here, the user can choose which transition to take next in the given process:
```

Comment:
er det det bilelderne nedenudenr viser? måske lige skriv explicit som fx in the given process as shown on autoref til figurerne